### PR TITLE
Add edge case handling and tests for initConsistentUtreexoState

### PR DESCRIPTION
**Pull request by 1amhesus**
### Why this change is needed

The `initConsistentUtreexoState` function lacked explicit handling for some edge cases, such as `tipHeight < 0` or `tipHash == nil`. This could lead to unexpected behaviors or errors during state reconstruction.

### What was changed

1. Explicit checks were added to `initConsistentUtreexoState` for:
   - `tipHeight < 0`: The function now returns early when this condition is met.
   - `tipHash == nil`: The function now returns an error for this invalid state.
   
2. New test cases were added to validate these edge conditions:
   - Ensures the function behaves correctly when `tipHeight < 0`.
   - Validates error handling when `tipHash` is nil.

3. Wrapper functions were introduced in `indexers_test.go` to support reusable test chain setups.

### CODE REVIEW

File: blockchain/indexers/utreexobackend.go [hunk 1]
```diff
@@ -586,6 +586,15 @@ func upgradeUtreexoState(cfg *UtreexoConfig, p *utreexo.MapPollard,
 func (us *UtreexoState) initConsistentUtreexoState(chain *blockchain.BlockChain,
 	savedHash, tipHash *chainhash.Hash, tipHeight int32) error {
 
+	// Handle nil tipHash
```
Comment (kcalvinalvin) [inline blockchain/indexers/utreexobackend.go:589]: We handle both of the cases literally just below at line 600.

**Comment by kcalvinalvin** [comment]
This isn't needed since both of the said reasons why this change is needed is handle here

https://github.com/utreexo/utreexod/blob/03ed304b03dde9d28631add17b64ba7581eb7f49/blockchain/indexers/utreexobackend.go#L589-L593

The tests aren't relevant either because we rely on the fact that the height is negative and the savedHash is nil to see if the user is running a fresh node.

Any other reasons? If not, I'll be closing this PR.

File: blockchain/indexers/utreexobackend.go (hunk 1 above)
Comment (1amhesus) [reply blockchain/indexers/utreexobackend.go:589]: My intention was to handle cases where either `tipHash == nil` or `tipHeight < 0` happens on its own, which the logic at line 600 doesn’t cover.

File: blockchain/indexers/utreexobackend.go [hunk 2]
```diff
@@ -586,6 +586,15 @@ func upgradeUtreexoState(cfg *UtreexoConfig, p *utreexo.MapPollard,
...
+	if tipHash == nil {
+		return fmt.Errorf("tipHash is nil, cannot initialize Utreexo state")
+	}
+
+	// Handle negative tipHeight
+	if tipHeight < 0 {
+		return nil
+	}
```
Comment (1amhesus) [inline blockchain/indexers/utreexobackend.go:597]: `tipHeight == -1 && tipHash.IsEqual(&empty)` handles the initialization state when both tipHeight == -1 and tipHash == nil are true. However, I intended to handle tipHash == nil or tipHeight < 0 individually. While focusing solely on the "fresh node" scenario makes sense, explicitly handling tipHash == nil in other cases could improve stability and robustness.
